### ISSUE TITLE
Sonar JS version 5.0 incompatible with JDK10

### ISSUE THREAD

**Issue by rauset** [SonarSource/SonarJS#1110]
We are analyzing our code using SonarJS plugin. We have build our code using JDK10. We have upgraded Sonar JS to latest version 5.0 on our sonar instance but still we can see following warning during analyzing

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by net.sf.cglib.core.ReflectUtils$1 (file:/home/go/.sonar/cache/1bee17e3bd3f033c19378cf80e76e18b/sonar-javascript-plugin-5.0.0.6962.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of net.sf.cglib.core.ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

It would be great if this can be fixed or if there is any workaround for the same please let me know.

**Comment by rauset** [comment]
Hello,

But due to this warning our analyzing is affected at the end we are getting some weird error like this 

```
> Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.5.0.1254:sonar
>  While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
```

And nothing has been changed just we are using JDk 10 for building code.

**Comment by saberduck** [comment]
I am quite sure that your issue is not coming from the warning. Can you please sent full output of the analysis with debug enabled? `mvn -X ...`

**Comment by rauset** [comment]
i think you are correct it could be possible reason that we are using latest LTS version of Sonar which seems incompatible. found one post related to it https://github.com/spotbugs/sonar-findbugs/issues/184

We already have latest version of findbugs but not latest version of Sonar Community Edition. I wil ltry to upgrade to latest version of Sonar and then will get back on this post if it is not working.

**Comment by rauset** [comment]
Yes that would be great if you can do that

Thanks for your help.

**Comment by saberduck** [comment]
@rauset just create new topic on the community forum. Don't forget to include all relevant info - logs, versions, installed plugins, etc.
